### PR TITLE
Cleanup ElfLoader and BootElf

### PR DESCRIPTION
**Pull request by phire**
Various cleanups.

I originally went in to remove a GetPointer, but this code has been hardly touched since the original Open Sourcing, so I couldn't resist cleaning up a lot of other things while I was there.

### CODE REVIEW

**Comment by phire** [comment]
Rebased. This can now be merged.

delroth: the crash was fixed.

File: Source/Core/Core/Boot/Boot.cpp [hunk 1]
```diff
@@ -324,58 +324,35 @@ bool CBoot::BootUp()
...
-			Load_FST(elfWii);
-			Boot_ELF(_StartupPara.m_strFilename);
+		// Poor man's bootup
+		if(_StartupPara.bWii)
+			SetupWiiMemory(DiscIO::IVolume::COUNTRY_UNKNOWN);
+		else
+			EmulatedBS2_GC(true);
+
+		Load_FST(_StartupPara.bWii);
+		if(!Boot_ELF(_StartupPara.m_strFilename)) {
```
Comment (lioncash) [inline Source/Core/Core/Boot/Boot.cpp]: Brace on next line

Comment (Sonicadvance1) [inline Source/Core/Core/Boot/Boot.cpp]: Curly brace on the next line or no curly braces.

**Comment by phire** [comment]
Braces fixed.

**Pull request merged**